src/formatting/items.rs

```diff
@@ -1131,30 +1131,42 @@ pub(crate) fn format_trait(
         );
         result.push_str(&header);
 
+        // FIXME: rustfmt fails to format when there are comments before the ident.
+        if contains_comment(context.snippet(mk_sp(item.span.lo(), generics.span.lo()))) {
+            return None;
+        }
+
```

Comment:
Nit: if we're going to have an early bail (which is a good idea btw!) then i think we should adjust this a bit. 

The inline comment hi end is the beginning of generics, not the ident so if we can handle any comments between the ident and wherever the generics begin then let's reduce the end of this span we're using for the early bail.

Additionally, we can shift this check up front, as there's really no need for the allocation and formatting of the visibility and friends if we're going to have to bail out down here anyway